stripe/src/main/java/com/stripe/android/view/CountryAutoCompleteTextView.kt

```diff
@@ -5,52 +5,96 @@ import android.util.AttributeSet
+ * which configures a [CountryAdapter] to display list of countries in its popup.
+ *
+ * The style of [AutoCompleteTextView] can be changed via [R.styleable.CountryAutoCompleteTextView_countryAutoCompleteStyle],
+ * the style of popup items can be changed via [R.styleable.CountryAutoCompleteTextView_countryItemLayout].
+ */
 internal class CountryAutoCompleteTextView @JvmOverloads constructor(
     context: Context,
     attrs: AttributeSet? = null,
-    defStyleAttr: Int = 0
-) : FrameLayout(context, attrs, defStyleAttr) {
-    private val viewBinding =
-        CountryAutocompleteViewBinding.inflate(
-            LayoutInflater.from(context),
-            this
-        )
-    private val countryTextInputLayout = viewBinding.countryTextInputLayout
+    defStyleAttr: Int = com.google.android.material.R.attr.textInputStyle
+) : TextInputLayout(context, attrs, defStyleAttr) {
 
-    private val layoutInflater = LayoutInflater.from(context)
-    private val countryAdapter = CountryAdapter(
-        context,
-        CountryUtils.getOrderedCountries(
-            ConfigurationCompat.getLocales(context.resources.configuration)[0]
-        )
-    ) {
-        CountryTextViewBinding.inflate(
-            layoutInflater,
-            it,
-            false
-        ).root
-    }
+    @StyleRes
+    private var countryAutoCompleteStyleRes: Int = INVALID_COUNTRY_AUTO_COMPLETE_STYLE
+
+    @LayoutRes
+    private var countryLineLayoutRes: Int = DEFAULT_COUNTRY_LINE_LAYOUT
 
     @VisibleForTesting
-    internal val countryAutocomplete = viewBinding.countryAutocomplete
+    internal val countryAutocomplete: AutoCompleteTextView
 
     /**
      * The 2 digit country code of the country selected by this input.
      */
     @VisibleForTesting
-    var selectedCountry: Country?
+    var selectedCountry: Country? = null
+        set(value) {
+            field = value
+            value?.let {
+                countryChangeCallback(it)
+            }
+        }
 
     @JvmSynthetic
     internal var countryChangeCallback: (Country) -> Unit = {}
 
+    private var countryAdapter: CountryAdapter
+
     init {
+        context.withStyledAttributes(
+            attrs,
+            R.styleable.CountryAutoCompleteTextView
+        ) {
+            countryAutoCompleteStyleRes = getResourceId(
+                R.styleable.CountryAutoCompleteTextView_countryAutoCompleteStyle,
+                INVALID_COUNTRY_AUTO_COMPLETE_STYLE
+            )
+
+            countryLineLayoutRes = getResourceId(
+                R.styleable.CountryAutoCompleteTextView_countryItemLayout,
+                DEFAULT_COUNTRY_LINE_LAYOUT
+            )
+        }
+
+        countryAutocomplete = AutoCompleteTextView(
+            context,
+            null,
+            R.attr.autoCompleteTextViewStyle
+        ).takeIf { countryAutoCompleteStyleRes == INVALID_COUNTRY_AUTO_COMPLETE_STYLE }
+            ?: AutoCompleteTextView(context, null, 0, countryAutoCompleteStyleRes)
```

Comment:
Let's move this into a separate method that explains the differences between the two paths. Also I'd prefer not using `takeIf` here because it creates an unnecessary object.